src/server/project.js

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+var fs = require('fs'),
+    path = require('path'),
+    cordovaServe = require('cordova-serve'),
+    Q = require('q'),
+    dirs = require('./dirs'),
+    pluginUtil = require('./utils/plugins'),
+    prepareUtil = require('./utils/prepare'),
+    utils = require('./utils/jsUtils');
+
+/**
+ * @constructor
+ */
+function Project(simulator, platform, debugHostHandlers) {
+    this._simulator = simulator;
+    this.platform = platform;
+    this.debugHostHandlers = debugHostHandlers;
```

Comment:
I notice that `this.debugHostHandlers` seems static. The intent is that if a debug-host disconnects from the server, this should be updated to properly reflect that there is no longer any debug-host, so on the next prepare / refresh, the debug-specific panels will not be added to sim-host. Does the current `Project` class handle debug-hosts disconnecting?